sdk/core/azure-core-serializer-json-gson/src/main/java/com/azure/core/serializer/json/gson/GsonJsonSerializer.java

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.serializer.json.gson;
+
+import com.azure.core.serializer.JsonSerializer;
+import com.azure.core.util.CoreUtils;
+import com.google.gson.Gson;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+/**
+ * GSON based implementation of the {@link JsonSerializer} interface.
+ */
+public final class GsonJsonSerializer implements JsonSerializer {
+    private final Gson gson;
+
+    /**
+     * Constructs a {@link JsonSerializer} using the passed {@link Gson} serializer.
+     *
+     * @param gson Configured {@link Gson} serializer.
+     */
+    GsonJsonSerializer(Gson gson) {
+        this.gson = gson;
+    }
+
+    @Override
+    public <T> T deserialize(byte[] input, Class<T> clazz) {
+        return gson.fromJson(CoreUtils.bomAwareToString(input, null), clazz);
+    }
+
+    @Override
+    public byte[] serialize(Object value) {
+        return gson.toJson(value).getBytes(StandardCharsets.UTF_8);
+    }
```

Comment:
Should serialize also support other charsets instead of always using UTF 8?

Reply:
I believe it is safe to default to using UTF-8 when dealing with JSON, looking around in other places in code `HttpRequest` defaults to using `UTF-8` when sending `String` bodies, so at minimum this will fall in line with that.